assets/js/components/Sectors.vue

```diff
@@ -378,28 +407,32 @@ export default Vue.extend({
 	    label_spacing = $this.label_spacing;
 
       // draw the legend
-       const label = $this.svg.select("g.legend")
-	  .selectAll(".label")
-      // we can access root directly here because we don't care about partitioning (yet?)
-	  .data($this.getRoot().descendants().slice(1))
-	  .enter().append("g")
-	  .attr("id", $this.getLabelID)
-	  .attr("class", "label")
-    .attr("opacity", (d) => d.depth == 1 ? 1 : 0)
-    .style("display", (d) => d.depth == 1 ? "inline" : "none")
-	  .attr("transform", function(d, i) {
-      let y = (label_size + label_spacing) * i;
-      return `translate(0,${y})`
-    });
+      const label = $this.svg.select(".legend")
+      .selectAll(".label")
+        // we can access root directly here because we don't care about partitioning (yet?)
+  	  .data($this.getRoot().descendants().slice(1))
+  	  .enter().append("xhtml:div")
+  	  .attr("id", $this.getLabelID)
+  	  .attr("class", "label")
+      .style('white-space','nowrap')
+      .style("opacity", (d) => d.depth == 1 ? 1 : 0)
+      .style("display", (d) => d.depth == 1 ? "block" : "none");
+
       this._labels = label;
-      label.append("rect")
-	.attr("width", label_size)
-	.attr("height", label_size)
-	.attr("fill", $this._colour)
-      label.append('text')
-	.attr('x', label_size + label_spacing)
-	.attr('y', label_size)
-	.text(function(d) { return d.data.name; });
+      label.append("xhtml:span")
+      .style('width', label_size + 'px')
```

Comment:
wherever you have .style() for html, things can be done so much prettier.
actually, it applies to most of the svg too probably.

do a test: see if you can use {{ variables }} inside vue's style block.